apps/pipelines/nodes/nodes.py

```diff
@@ -148,40 +213,54 @@ def _create_model_from_data(value_data, model_name: str):
         return schema
 
 
-class UpdateParticipantMemory(PipelineNode):
-    """A simple component to merge the the input data into the participant's memory data. If key_name is specified
-    the input data will be merged with `key_name` as the key name.
-    """
+class ExtractStructuredData(ExtractStructuredDataNodeMixin, LLMResponse):
+    __human_name__ = "Extract Structured Data"
+    data_schema: str
+
 
-    __human_name__ = "Update participant memory"
+class ExtractParticipantData(ExtractStructuredDataNodeMixin, LLMResponse):
+    __human_name__ = "Extract Participant Data"
+    data_schema: str
     key_name: str | None = None
 
-    def _process(self, state: PipelineState) -> PipelineState:
-        """Strategy:
-        - If the input is a string or list, a `key_name` must be present to be valid JSON.
-        - If `key_name` is present, we create a new dictionary with `input` as the value to `key_name`.
-        - Merge this new dictionary with the current participant data, essentially overriding existing data
-        """
-        new_data = state["messages"][-1]
+    def get_json_schema(self):
+        return ExtractParticipantData.to_json_schema(json.loads(self.data_schema))
+
+    def get_reference_data(self, state) -> dict:
         session = state["experiment_session"]
+        data = session.get_participant_data()
+        if self.key_name:
+            return data.get(self.key_name, {})
```

Comment:
similar to above re empty string vs dict
```suggestion
            return data.get(self.key_name, "")
```